.github/actions/main.ts

```diff
@@ -38,7 +38,7 @@ rl.on('line', function (line) {
 rl.on('close', async function () {
   const result = await findLastGoodBuild(
     shas,
-    process.env.BRANCH,
+    process.env.BRANCH || process.env.GIT_BRANCH,
```

Comment:
This is basically a case of "Computer, do what I want you to do, not what I tell you to do."